File: edgegate-node/test/config.test.js
import test from 'node:test';
import assert from 'node:assert/strict';
import { validateConfig } from '../src/config.js';

test('validates a route configuration', () => {
  const config = { routes: [{ path: '/api', backends: ['http://localhost:9000'] }] };
  assert.doesNotThrow(() => validateConfig(config));
  assert.equal(config.routes[0].strategy, 'round_robin');
});

test('rejects duplicate paths and invalid strategies', () => {
  assert.throws(() => validateConfig({ routes: [
    { path: '/api', backends: ['http://localhost:9000'] },
    { path: '/api', backends: ['http://localhost:9001'] }
  ] }), /duplicate route/);
  assert.throws(() => validateConfig({ routes: [
    { path: '/api', backends: ['http://localhost:9000'], strategy: 'random' }
  ] }), /unsupported strategy/);
});

File: edgegate/internal/config/config_test.go
package config

import "testing"

func TestValidate(t *testing.T) {
	cases := []struct {
		name string
		cfg  Config
		want bool
	}{
		{"valid", Config{Routes: []Route{{Path: "/api", Backends: []string{"http://localhost:8081"}}}}, true},
		{"missing path", Config{Routes: []Route{{Backends: []string{"http://localhost:8081"}}}}, false},
		{"invalid backend", Config{Routes: []Route{{Path: "/api", Backends: []string{"localhost:8081"}}}}, false},
		{"invalid strategy", Config{Routes: []Route{{Path: "/api", Backends: []string{"http://localhost:8081"}, Strategy: "random"}}}, false},
	}
	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			if got := Validate(tc.cfg) == nil; got != tc.want {
				t.Fatalf("valid=%v, want %v", got, tc.want)
			}
		})
	}
}
